package/MDAnalysis/converters/OpenMMParser.py

```diff
@@ -102,42 +105,81 @@ def _mda_topology_from_omm_topology(self, omm_topology):
         resnames = [r.name for r in omm_topology.residues()]
         resids = [r.index + 1 for r in omm_topology.residues()]
         resnums = resids.copy()
         segids = [str(c.index) for c in omm_topology.chains()]
         bonds = [(b.atom1.index, b.atom2.index) for b in omm_topology.bonds()]
         bond_orders = [b.order for b in omm_topology.bonds()]
         bond_types = [b.type for b in omm_topology.bonds()]
-
-        n_atoms = len(atomids)
-        n_residues = len(resids)
-        n_segments = len(segids)
+        elements = [None if a.element is None else a.element.symbol
+                    for a in omm_topology.atoms()]
 
         attrs = [
             Atomids(np.array(atomids, dtype=np.int32)),
             Atomnames(np.array(atomnames, dtype=object)),
-            Atomtypes(np.array(atomtypes, dtype=object)),
             Bonds(bonds, types=bond_types, order=bond_orders, guessed=False),
             ChainIDs(np.array(chainids, dtype=object)),
-            Elements(np.array(elements, dtype=object)),
-            Masses(np.array(masses, dtype=np.float32)),
             Resids(resids),
             Resnums(resnums),
             Resnames(resnames),
             Segids(segids),
         ]
 
+        if not any(elements):
+            atomtypes = guess_types(atomnames)
+            attrs.append(Atomtypes(np.array(atomtypes, dtype=object)))
+            masses = guess_masses(atomtypes)
+            attrs.append(Masses(np.array(masses)))
+            warnings.warn("Element information is missing, elements "
+                          "attribute will not be populated. "
+                          "Atomtype attribute will be guessed using atom "
+                          "name and mass will be guessed using atomtype."
+                          "See MDAnalysis.topology.guessers.")
+        else:
+            validated_elements = []
+            masses = []
+            for a in omm_topology.atoms():
+                elem = a.element
+                if elem is not None and elem.symbol.capitalize() in SYMB2Z:
+                    validated_elements.append(a.element.symbol.capitalize())
+                    masses.append(a.element.mass._value)
+                else:
+                    wmsg = (f"Unknown element {elem} found for some atoms. "
+                            f"These have been given an empty element record "
+                            f"with their atomtype set to 'X' "
+                            f"and their mass set to 0.0. "
+                            f"If needed they can be guessed using "
+                            f"MDAnalysis.topology.guessers.")
+                    warnings.warn(wmsg)
+                    validated_elements.append('')
+                    masses.append(0.0)
+
+            attrs.append(Elements(np.array(validated_elements, dtype=object)))
+            attrs.append(Masses(np.array(masses)))
+            atomtypes = [elem if len(elem) != 0 else 'X'
```

Comment:
One of my favourite things about Python is [truthiness](https://www.freecodecamp.org/news/truthy-and-falsy-values-in-python/)! `if len(elem) != 0` and `if elem` accomplish pretty much the same thing.